config.go

```diff
@@ -546,6 +564,30 @@ func loadConfig() (*config, []string, error) {
 			return nil, nil, err
 		}
 
+		// Ensure pool fee is valid
+		if cfg.PoolFee < 0 || cfg.PoolFee > 1 {
+			str := "%s: poolfee should be between 0 and 1"
+			err := fmt.Errorf(str, funcName)
+			return nil, nil, err
+		}
+
+		// Ensure the password to unlock the wallet.
+		// Needed to pay dividends to pool contributors.
+		if cfg.WalletPass == "" {
+			str := "%s: the walletpass option is not set"
+			err := fmt.Errorf(str, funcName)
+			return nil, nil, err
+		}
+
+		// Ensure address to collect pool fees is provided.
+		if len(cfg.PoolFeeAddrs[0]) == 0 {
```

Comment:
We can refine this some more, this should only be set when not in solo pool mode.

Reply:
Also this is taking the length of the first pool fee address, not the length of addresses provided.

Reply:
in light of the issue discovered with `jessevdk/go-flags` please add comments explaining why things are being done this way for future reference.